cpp/tests/io/comp/comp_test.cpp

```diff
@@ -49,13 +45,39 @@ namespace nvcomp = cudf::io::detail::nvcomp;
  * whose interface and setup is different for each codec.
  */
 template <typename Decompressor>
-struct DecompressTest : public cudf::test::BaseFixture {
-  void Decompress(std::vector<uint8_t>& decompressed,
-                  uint8_t const* compressed,
-                  size_t compressed_size)
+struct DecompressTest : public cudf::test::BaseFixture, public testing::WithParamInterface<hw> {
+  [[nodiscard]] std::vector<uint8_t> vector_from_string(std::string const str) const
```

Comment:
Oh no, this makes a new, local string then you return pointers to that local one which will be destroyed after exiting the function.